internal/disabler/disabler.go

```diff
@@ -0,0 +1,111 @@
+// package disabler contains a component that can be enabled and disabled dynamically
+package disabler
+
+import (
+	"context"
+	"errors"
+
+	"github.com/pomerium/pomerium/internal/log"
+)
+
+var errDisabled = errors.New("disabled")
+
+type Handler interface {
+	RunEnabled(ctx context.Context) error
+}
+
+type Disabler interface {
+	Run(ctx context.Context) error
+	Enable()
+	Disable()
+}
+
+type disabler struct {
+	name            string
+	handler         Handler
+	onChangeEnabled chan bool
+}
+
+// New creates a new Disabler. When the Disabler is enabled, the Handler's
```

Comment:
I'd put in a vote for naming this Enabler rather than Disabler. The phrase "When the Disabler is enabled" makes my head hurt 🙂.